### ISSUE TITLE
Exclude live videos

### ISSUE THREAD

**Issue by 726a67** [trevorsharp/youtube-podcast-feeds#20]
Currently, feeds will include both live and fixed-length videos. Live videos (and especially 'upcoming' live videos) can produce some awkward podcast client interactions (e.g. when automatic downloads are configured and the content url doesn't yet point to actual content).

It would be great if feeds could be configured to ignore/exclude live video streams (perhaps leveraging something like yt-dlp's 'is_live' field).

**Comment by trevorsharp** [comment]
Currently to avoid this, videos are only added to the feed if YouTube’s upload status is “processed”. I’ve found that usually this prevents live videos from being added to the feed until the video has finished and is fully processed by YouTube.

I’m curious if you have an example channel or playlist that I could look at to see what might be going on. Also, is this for live videos or video premieres? I could imagine that there might be some issues around premieres as I haven’t tested that.

**Comment by trevorsharp** [comment]
I’ve added an improvement to the filtering which should exclude any active or upcoming live videos. The expected behavior is that a live video will be added only after it has completed, has been processed, and is accessible as a VOD. Pull the updated docker image at `trevorsharp/youtube-podcast-feeds:latest` and let me know if you still see issues with live videos.

**Comment by 726a67** [comment]
Unfortunately none of my feeds currently have an upcoming/premiering video to test against, but I expect at least one of them will release a candidate video in the next week or so.

I've updated to the latest image and will report back with my findings.

Thanks for the quick turnaround!

**Comment by 726a67** [comment]
I had some free time so I found a channel with a premiering video and used it to build a new test feed:

Channel: https://www.youtube.com/channel/UCLA_DiR1FfKNvjuUpBHmylQ
Video: https://www.youtube.com/watch?v=2W3P0Utvdbs

I built a second feed based on a playlist comprised of entirely premiering videos:

https://www.youtube.com/playlist?list=PLU12uITxBEPHHlOIWGAIezbshH82rGpKp

In both cases, the feeds correctly ignore/exclude the premiere videos. As far as I'm concerned, this issue is resolved. Thanks again.
